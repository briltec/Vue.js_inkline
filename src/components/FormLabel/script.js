import ClassesProviderMixin from 'inkline/mixins/components/providers/ClassesProviderMixin';
import InjectParentFormProviderMixin from 'inkline/mixins/forms/providers/InjectParentFormProviderMixin';

import SizePropertyMixin from 'inkline/mixins/components/properties/SizePropertyMixin';

export default {
    name: 'IFormLabel',
    mixins: [
        ClassesProviderMixin,
        InjectParentFormProviderMixin,

        SizePropertyMixin,
    ],
    props: {
        placement: {
            type: String,
            default: 'default'
        }
    },
    methods: {
        focusInput() {
            const siblings = this.$parent.$children;
            const selfIndex = siblings.indexOf(this);

            const input = siblings.find(
                (sibling, index) => index > selfIndex && sibling.hasOwnProperty('focusInputRef'));

            if (input) {
                input.focusInputRef();
            }
        }
    },
    created() {
        this.classesProvider.add(() => ({
            [`-${this.placement}`]: this.placement !== 'default'
        }));
    }
};
