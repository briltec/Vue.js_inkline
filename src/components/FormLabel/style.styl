@import '../../css/config'
@import '../../css/mixins'
@import '../Input/css/_variables'
@import '../Input/css/_mixins'
@import 'css/_sizes'

/**
 * Form Label
 */
.form-label
    position: relative
    flex-basis: 100%

    &.-left
    &.-right
        display: flex
        flex: 0 0 auto
        align-self: center
        margin-bottom: 0

    &.-right
        order: 1


    @extend .form-label.-md
