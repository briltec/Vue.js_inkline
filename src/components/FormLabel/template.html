<label class="form-label" @click="focusInputRef" :class="classes">
    <slot name="label"></slot>
</label>