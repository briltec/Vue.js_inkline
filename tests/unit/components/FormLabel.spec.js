import { mount } from '@vue/test-utils';
import FormLabel from 'inkline/components/FormLabel';

describe('Components', () => {
    describe('FormLabel', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = mount(FormLabel);
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
