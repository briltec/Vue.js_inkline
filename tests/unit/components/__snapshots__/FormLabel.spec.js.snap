// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components FormLabel should render correctly 1`] = `<label class="form-label"></label>`;

exports[`Components FormLabel should render correctly 2`] = `<label class="form-label"></label>`;
